runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -111,6 +115,26 @@ TR::SymbolValidationManager::SymbolValidationManager(TR::Region &region, TR_Reso
       _alreadyGeneratedRecords.insert(
          new (_region) ArrayClassFromComponentClassRecord(arrayClass, component));
       }
+
+#if defined(J9VM_OPT_JITSERVER)
+   if (TR::CompilationInfo::get()->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
+      {
+      // If needed, populate the client session with classes not worth remembering
+      auto clientData = _fej9->_compInfoPT->getClientData();
+      auto &classesNotWorthRemembering = clientData->getSystemClassesNotWorthRemembering();
+         {
+         OMR::CriticalSection classesNotWorthRememberingMonitor(clientData->getClassesNotWorthRememberingMonitor());
+         if (classesNotWorthRemembering.size() != SYSTEM_CLASSES_NOT_WORTH_REMEMBERING_COUNT)
+            {
+            for (int i = 0; i < SYSTEM_CLASSES_NOT_WORTH_REMEMBERING_COUNT; ++i)
+               {
+               auto classNotWorthRemembering = &_systemClassesNotWorthRemembering[i];
+               classesNotWorthRemembering.push_back({ classNotWorthRemembering->_className, NULL, classNotWorthRemembering->_checkIsSuperClass });
+               }
+            }
+         }
+      }
```

Comment:
I think we should execute this block of code in the constructor of a session where our new vector of classes not worth remembering is defined. This way we can avoid locking getClassesNotWorthRememberingMonitor() every time we create an SVM object. I am proposing to take this block of code (more or less) and create a static function in `SymbolValidationManager`. Then call this function in the constructor of a client session. We may not even need a vector, just an array of structs should be enough, though I don't know if we can use the computed number of elements in another file.

Reply:
Yeah, I will implement that. I think we don't even need a monitor if array initialization is done in client session constructor.